Exec/Make.auto_source

```diff
@@ -39,27 +39,27 @@ $(CASTRO_AUTO_SOURCE_DIR)/extern.F90: $(EXTERN_PARAMETERS) $(EXTERN_TEMPLATE)
 #------------------------------------------------------------------------------
 
 # runtime parameter support for extern/ routines
-F90EXE_sources += prob_params_auto.F90
 CEXE_sources += prob_parameters.cpp
-FEXE_headers += prob_parameters_F.H
 CEXE_headers += prob_parameters.H
 
-$(CASTRO_AUTO_SOURCE_DIR)/prob_parameters.cpp: $(CASTRO_AUTO_SOURCE_DIR)/prob_params_auto.F90
-
-PROB_PARAMS_TEMPLATE := $(TOP)/Util/scripts/prob_params.template
 # search Bdirs for _prob_params files
 PROB_PARAMETERS := $(foreach dir, $(Bdirs), $(wildcard $(TOP)/$(dir)/_prob_params))
 # the problem's own parameters always come at the end so they override
 # any others
 PROB_PARAMETERS += $(wildcard $(PROBLEM_DIR)/_prob_params)
 
+$(CASTRO_AUTO_SOURCE_DIR)/prob_parameters.cpp: $(CASTRO_AUTO_SOURCE_DIR)/prob_parameters.H
+
+# for dependency resolution
+$(depEXETempDir)/%.d: $(CASTRO_AUTO_SOURCE_DIR)/prob_parameters.H
```

Comment:
Why is this needed?

Reply:
we have to force the build system to create the header file before it does the C++ dependency resolution, otherwise it will fail, because it can't find the prob_parameters.H.